apps/provider.c

```diff
@@ -18,7 +18,7 @@
 #include <openssl/safestack.h>
 #include <openssl/provider.h>
 #include <openssl/core.h>
-#include <openssl/core_numbers.h>
+#include <openssl/core_functions.h>
```

Comment:
This name doesn't translate well in my mind.  While that header does include function typedefs for core functions offered to the provider, the majority of it includes function typedefs that *providers* may offer.

The reason I chose "core_numbers" was that I viewed the numbers as the primary data and the associated typedefs as auxiliary data at the time, and those numbers *are* offered by the core.  Not sure I'd view it as categorically today, though, but my imagination hasn't been enough to find a new name that I'm satisfied with...

Reply:
> This name doesn't translate well in my mind.

Well, I anticipated that this would be the most controversial change, so I left it in a separate commit from the beginning, to be able to change or drop it. :-)

I understand your concerns that the term "functions" is too narrowed down for what the header contains. But "numbers" is quite the opposite, it is vague and leaves a lot of room for imagination, too. After all, the fact that we are using natural numbers for representing the IDs of the functions in the dispatch table is an implementation detail, isn't it?

Speeking of dispatch tables: how would the name `<openssl/core_dispatch.h> translate in your mind`? Would that fit better? If you like the name, then I would suggest moving the declaration of `struct ossl_dispatch_st` from core.h into core_dispatch.h as well, then the name makes even more sense.

https://github.com/openssl/openssl/blob/fe3c5424ec5dffbeca5d71164aff3eb9a8cc008c/include/openssl/core.h#L33-L42

Reply:
Hmmm, `core_dispatch.h` is ok, but I would prefer to define the central types in `core.h`

Reply:
Done, see [8a94c5ed1](https://github.com/openssl/openssl/pull/12222/commits/8a94c5ed1e78bb6db0079b722dce0151cf2b5e27).